presto-geospatial/src/test/java/com/facebook/presto/geospatial/TestGeoQueries.java

```diff
@@ -0,0 +1,326 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.geospatial;
+
+import com.facebook.presto.Session;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.testing.LocalQueryRunner;
+import com.facebook.presto.tests.AbstractTestQueryFramework;
+import com.facebook.presto.tpch.TpchConnectorFactory;
+import com.google.common.collect.ImmutableMap;
+import org.testng.annotations.Test;
+
+import static com.facebook.presto.metadata.FunctionExtractor.extractFunctions;
+import static com.facebook.presto.testing.TestingSession.testSessionBuilder;
+import static com.facebook.presto.tpch.TpchMetadata.TINY_SCHEMA_NAME;
+
+public class TestGeoQueries
+        extends AbstractTestQueryFramework
+{
+    public TestGeoQueries()
+    {
+        super(TestGeoQueries::createLocalQueryRunner);
+    }
+
+    @Test
+    public void testSTPoint()
+            throws Exception
+    {
+        assertQuery("select st_as_text(st_point(1, 4))", "select 'POINT (1 4)'");
+    }
+
+    @Test
+    public void testSTLineString()
+            throws Exception
+    {
+        assertQuery("select st_as_text(st_line_string('linestring(1 1, 2 2, 1 3)'))", "select 'LINESTRING (1 1, 2 2, 1 3)'");
+    }
+
+    @Test
+    public void testSTPolygon()
+            throws Exception
+    {
+        assertQuery("select st_as_text(st_polygon('polygon ((1 1, 1 4, 4 4, 4 1))'))", "select 'POLYGON ((1 1, 4 1, 4 4, 1 4, 1 1))'");
+    }
+
+    @Test
+    public void testSTArea()
+            throws Exception
+    {
+        assertQuery("select st_area(st_geometry_from_text('polygon ((2 2, 2 6, 6 6, 6 2))'))", "select 16.0");
+    }
+
+    @Test
+    public void testSTCentroid()
+            throws Exception
+    {
+        assertQuery("select st_as_text(st_centroid(st_geometry_from_text('polygon ((1 1, 1 4, 4 4, 4 1))')))", "select 'POINT (2.5 2.5)'");
```

Comment:
nit: consider testing st_centroid on a non-polygon shape

ditto other methods which apply to a restricted set of shape types